kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/activities/metadata/ActivityMetadata.kt

```diff
@@ -61,7 +62,25 @@ internal class ActivityMetadata(
                     v.width,
                     v.height
                 )
+            }.toMutableList()
+
+        val stringsFromToolbarLayouts = TreeIterables.depthFirstViewTraversal(decorView)
```

Comment:
Can we avoid traversing the hierarchy twice? For example:
TreeIterables.depthFirstViewTraversal(decorView)
   .filter { it.visibility == View.VISIBLE }
   .filter( it is TextView || it is CollapsingToolbarLayout)
   .map { case for TextView and CollapsingToolbarLayout } ...

Reply:
done